wrappers/golang/curves/bn254/curve.go

```diff
@@ -0,0 +1,167 @@
+package bn254
+
+// #cgo CFLAGS: -I./include/
+// #include "curve.h"
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/core"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/runtime"
+)
+
+type Projective struct {
+	X, Y, Z BaseField
+}
+
+func (p Projective) Size() int {
+	return p.X.Size() * 3
+}
+
+func (p Projective) AsPointer() *uint32 {
+	return p.X.AsPointer()
+}
+
+func (p *Projective) Zero() Projective {
+	p.X.Zero()
+	p.Y.One()
+	p.Z.Zero()
+
+	return *p
+}
+
+func (p *Projective) FromLimbs(x, y, z []uint32) Projective {
+	p.X.FromLimbs(x)
+	p.Y.FromLimbs(y)
+	p.Z.FromLimbs(z)
+
+	return *p
+}
+
+func (p *Projective) FromAffine(a Affine) Projective {
+	z := BaseField{}
+	z.One()
+
+	p.X = a.X
+	p.Y = a.Y
+	p.Z = z
+
+	return *p
+}
+
+func (p Projective) ProjectiveEq(p2 *Projective) bool {
+	cP := (*C.projective_t)(unsafe.Pointer(&p))
+	cP2 := (*C.projective_t)(unsafe.Pointer(&p2))
+	__ret := C.bn254_eq(cP, cP2)
+	return __ret == (C._Bool)(true)
+}
+
+func (p *Projective) ProjectiveToAffine() Affine {
+	var a Affine
+
+	cA := (*C.affine_t)(unsafe.Pointer(&a))
+	cP := (*C.projective_t)(unsafe.Pointer(&p))
```

Comment:
Since `p` is a `*Projective`, it should be passed directly
```suggestion
	cP := (*C.projective_t)(unsafe.Pointer(p))
```